tests/test_environment_context_awareness.py

```
@@ -0,0 +1,199 @@
+    
+    def test_eslint_json_parsing_with_warnings(self):
+        """Test parsing ESLint output that includes warnings before JSON."""
+        # Simulate the exact issue found in MCP framework testing
+        mock_output = """
+> mcp-url-knowledge-graph@0.1.0 lint
+> eslint src/**/*.ts
+
+[{"filePath":"/test/file.ts","messages":[{"ruleId":"no-undef","severity":2,"message":"'process' is not defined","line":1,"column":1}]}]
+        """
+        
+        from unittest.mock import Mock
```

Ruff (0.12.2)

Redefinition of unused `Mock` from line 19

Remove definition: `Mock`

(F811)